behave_reportportal/behave_agent.py

```diff
@@ -0,0 +1,253 @@
+"""Functionality for integration of Behave tests with Report Portal."""
+from functools import wraps
+
+from prettytable import PrettyTable
+from reportportal_client import ReportPortalService
+from reportportal_client.helpers import get_launch_sys_attrs
+from reportportal_client.service import _dict_to_payload
+
+from behave_reportportal.utils import get_package_version, timestamp
+
+
+def check_rp_enabled(func):
+    """Verify is RP is enabled in config."""
+
+    @wraps(func)
+    def wrap(*args, **kwargs):
+        if args and isinstance(args[0], BehaveAgent):
+            if not args[0]._rp:
+                return
+
+        func(*args, **kwargs)
+
+    return wrap
+
+
+def create_rp_service(cfg):
+    """Create instance of ReportPortalService."""
+    if not cfg.enabled:
+        return
+    return ReportPortalService(
+        endpoint=cfg.endpoint, project=cfg.project, token=cfg.token
```

Comment:
Actually it does not matter for me, but I prefer early return patter. From my point of view it makes code more linear and readable. What do you think?

Reply:
The function returns None by default. Logically, we need to create an instance only if reporting is enabled in config. In your variant, you check that reporting is disabled in order to return None.

Reply:
Both options do the same work. I just talk about readability

Reply:
No clue why your variant is more readable:)

Reply:
Because it makes code more linear. This code can have more if statements in future and it will have bigger nesting.
Since both options work the same I belive that it is matter of taste. I just can provide [link](https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement)  where we can see that a lot of people think the same. But anyway if it is some kind of project habbit - I will update it

Reply:
Provided link is not Python-related. Secondly, I do not understand what you mean by saying "linear". The fewer if statements the better. The less code the better. I'm not seeing any benefits in making reverse conditions in that place.

Reply:
ok, if it is critical. But this question is not language related. Golden path should be more readable. And argument that less code is better does not work in all cases for me

Reply:
This question is language-related. Python has its own coding philosophy compared to other languages. It has different implementations of design patterns. It has its own compiler that adds `return None` to functions.